src/views/ProductList.tsx

```diff
@@ -0,0 +1,42 @@
+import React, { useState, useEffect } from 'react'
+import ProductListItem from '../components/ProductListItem'
+import { Container, LinearProgress } from '@material-ui/core'
+import Product from '../domain/Product';
+import { getAll } from '../data/ProductService';
+
+
+function ProductList() {
+    const initialProducts = [] as Product[]
+    const [products, setProducts] = useState(initialProducts)
```

Comment:
Nu e nevoie de declararea initialProducts, poti face ca in fisierul de deasupra, eventual cu `useState<Product[]>([]);`